js/stdlib/ts/ALGO.ts

```diff
@@ -1309,7 +1309,9 @@ export const connectAccount = async (networkAccount: NetworkAccount): Promise<Ac
         debug('getState');
         return await getState_(getC, (vibna:BigNumber) => {
           if ( vibne.eq(vibna) ) { return vtys; }
-          throw Error(`Expected state ${vibne}, got ${vibna}`);
+          const expectedLoc = bin._stateSourceMap[vibne.toNumber().valueOf()];
+          const actualLoc = bin._stateSourceMap[vibne.toNumber().valueOf()];
+          throw Error(`Expected state ${vibne} at ${expectedLoc}, got ${vibna} at ${actualLoc}`);
```

Comment:
There's a function in shared_impl (or maybe backend... look at `assert`) for printing these things.

I prefer: "Expected the DApp to be at the commit() on line XYZ, but it is at the commit() on line ABC."

If you add the stack traces (which would require doing something to the `DLS_FromConsensus` statement) then you'd put them after... like "\nStack trace of expected: ...\nStack trace of actual: ..."